### PR TITLE
REDIS: Adding detection of not running instances

### PR DESCRIPTION
**Pull request by m3rlinux**

### CODE REVIEW

**Comment by jherbel** [comment]
Hi,
thanks for this PR, it has been queued for internal review. Please note that this will take a while.
Best, Jörg
Internal reference: CMK-7506

**Comment by m3rlinux** [comment]
Hi Jörg,
are there any news about this commit?
Thanks!

File: agents/plugins/mk_redis [hunk 1]
```diff
@@ -105,9 +105,13 @@ fi
 for INSTANCE in ${REDIS_INSTANCES[*]}; do
     redis_args $INSTANCE
     # print server section
-    echo "[[[$INSTANCE|${!HOST}|${!PORT}]]]"
```
Comment (jherbel) [inline agents/plugins/mk_redis:108]: I think this line has to stay

**Review by jherbel — CHANGES_REQUESTED** [review]
I took a closer look. Besides the one line-specific comment, I have some general remarks:

- We unfortunately cannot merge this in its current state. Any instance which is not running will now be detected and have CRIT as its monitoring status. I do not believe that this is generally desirable.
- It would be OK to make the detection of not-running instances optional. Furthermore, one would also have to make the monitoring status of not-running instances configurable. These two points (especially the first one) require modifications which should only be made after this check plugin has been ported to the new API. 
- Finally, we would definitely need to test that the parse and the check function can actually deal with the additional agent output. Again, this should only be done after porting the check (and the already existing tests) to the new API.

So, to conclude, if you really want to go along with this, you would have to first port this check plugin to the new API, including the existing tests and the parse function (in a separate PR). Then, you could go along with the actual modifications (new discovery ruleset, extending the check ruleset etc.).

**Comment by gurubert** [comment]
This fix with "pidof redis-server" does not work on hosts with multiple redis instances where a few are running and one or two not.

**Comment by jherbel** [comment]
Closed due to inactivity

**Pull request closed without merging**